routes/truckRoutes.js

```diff
@@ -1 +1,62 @@
-// put your Express router code in here
\ No newline at end of file
+var express = require( 'express' );
+var app = express();
+var trucks = require( '../trucks' );
+var router = express.Router();
+
+router.route( '/' )
+  .get( function( request, response ) {
+
+    var allTrucks = trucks.getTrucks();
+
+    response.send( allTrucks );
+  })
+  .post( function ( request, response ) {
+
+    var newTruck = request.body;
+
+    if ( newTruck ) {
+      trucks.addTruck( newTruck );
+      response.status( 201 ).send( newTruck );
+    } else {
+      response.status( 400 ).send( 'problem adding truck' );
+    }
+  });
+
+router.route( '/:name' )
+  .get( function( request, response ) {
+
+    var truckName = request.params.name;
+    var truck = trucks.getTruck( truckName );
+
+    if ( truck ) {
+      response.send( truck );
+    }else {
+      response.status( 404 ).json( 'truck not found' );
+    }
+  })
+  .delete( function( request, response ) {
+
+    var truckName = request.params.name;
+
+    trucks.removeTruck( truckName );
+    response.sendStatus( 200 );
+  });
+
+app.route( '/food-types' )
```

Comment:
This won't actually work here as there is no reference to `app` in this module. You would either have to leave this code in index.js or create a different router file to move the `/food-types` routes to.